src/index.js

```diff
@@ -0,0 +1,224 @@
+import { exec } from 'child_process';
+import { sync as findUp } from 'find-up';
+import * as fs from 'fs-extra';
+import globCb from 'glob';
+import * as path from 'path';
+import semver from 'semver';
+import * as zkSnark from 'snarkjs';
+import { promisify } from 'util';
+
+const glob = promisify(globCb);
+
+zkSnark.bigInt.prototype.toJSON = function() {
+  return this.toString();
+};
+
+const Extension = {
+  Json: '.json',
+  Circom: '.circom',
+  VkProof: '.vk_proof',
+  VkVerifier: '.vk_verifier',
+  Solidity: '.sol'
+};
+
+class Snarks {
+  constructor(embark) {
+    this.embark = embark;
+    this.circuitsConfig = embark.pluginConfig;
+    this.compiledCircuitsPath = embark.config.dappPath('.embark', 'snarks');
+    // v5 detection won't work as desired with "dev embark" (built in monorepo)
+    // until after first prerelease of embark v5; can assign `true` to test
+    // this.isEmbark5 = true;
+    this.isEmbark5 = semver(this.embark.version).major >= 5;
```

Comment:
Maybe instead of calling it `isEmbark5` we can call it something like `usesNewEventSystem` or something along those lines. Thing is that, most likely, this will still apply in e.g. Embark v6, 7, 8 ... etc as well.

Reply:
I could go either way on this one. I guess I'm less confident that the events system won't evolve in a breaking way when we get to Embark v6. We could use `usesNewEventSystem` for now, and in the future introduce `isEmbark5` where `usesNewEventSystem` then represents v6, and so on into the future.

Another option would be to generally recommend that plugin authors (including ourselves) publish new major versions of plugins, containing needed changes, as opposed to trying to make them multi-version compatible. Maybe that's the cleanest way. This `embark-snark` plugin was never published to the registry. We could publish version `1.0.0` compatible with Embark v4 and (when it's time) `2.0.0` compatible with v5. The `isEmbark5` branching could then be removed entirely. Thoughts?

Reply:
> Another option would be to generally recommend that plugin authors (including ourselves) publish new major versions of plugins, containing needed changes, as opposed to trying to make them multi-version compatible.

^ This. It'll also force us to strongly consider what we want to put under our umbrella as "officially supported". Given the available bandwidth, it's better for us to put most of our focus on making the core great and have other people maintain plugins on top of that.